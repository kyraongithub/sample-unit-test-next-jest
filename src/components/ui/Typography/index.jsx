import React from 'react';

const Typography = (props) => {
  const { children, type = 'h1' } = props;
  const Element = type;
  return <Element>{children}</Element>;
};

export default Typography;
