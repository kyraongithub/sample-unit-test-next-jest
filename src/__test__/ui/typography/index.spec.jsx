import Typography from '@/components/ui/Typography';
import { render } from '@testing-library/react';

describe('Typography Component', () => {
  it('should render typography', () => {
    const { container } = render(
      <Typography type="h1">hello world</Typography>
    );
    expect(container).toMatchSnapshot();
  });
});
