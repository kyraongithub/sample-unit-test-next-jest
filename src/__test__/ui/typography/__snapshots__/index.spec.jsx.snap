// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Component should render typography 1`] = `
<div>
  <h1>
    hello world
  </h1>
</div>
`;
